packages/pyright-internal/src/analyzer/typeUtils.ts

```diff
@@ -1053,9 +1053,35 @@ export function getTypeVarScopeIds(type: Type): TypeVarScopeId[] {
     return scopeIds;
 }
 
-// Specializes the class with "Unknown" type args (or the equivalent for ParamSpecs
-// or TypeVarTuples).
-export function specializeWithUnknownTypeArgs(type: ClassType, tupleClassType?: ClassType): ClassType {
+/**
+ * if the type we're narrowing is Any or Unknown, we don't want to specialize using the
+ * variance/bound for compatibility with less strictly typed code (cringe)
```

Comment:
I would like `Any` and `Unknown` to specialize using the variance/bound for my code.

But I understand that for people who don't report `Unknown`, it would be more noisy.

@DetachHead  What do you think of a new configuration option:
![strictGenericNarrowing](https://github.com/user-attachments/assets/4adf96d7-5950-492c-9c4a-aad693306cc6)

I think I could implement it.

Reply:
Or can we just implement the new narrowing with `Any` / `Unknown` and see how many people complain about it?
It would be a bit of a waste to have an option when no one is asking for it.

Reply:
the option sounds good cuz i want to maintain backwards compatibility for pyright users where possible